### PR TITLE
[wip] Feature/solana fixes

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: plugins/aea-ledger-solana/aea_ledger_solana/__init__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Python package wrapping the public and private key cryptography and ledger api of Solana."""
+
+from .solana import *  # noqa isort:skip
+from .solana import _IDL, _BYTECODE, _SOLANA  # noqa isort:skip
```
[the remark this answers is not in the document]
Comment (solarw) [reply plugins/aea-ledger-solana/aea_ledger_solana/__init__.py:24]: @angrybayblade same approach is used in other plugins (check ethereum one)
so just keep consistent

File: plugins/aea-ledger-solana/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+```bash
+docker run -d -p 8899:8899 -p 8900:8900 dassy23/solana-test-ledger:latest
+```
+
+```
+python setup.py install
+pip install open-aea[all]
+```
```
[the remark this answers is not in the document]
Comment (solarw) [reply plugins/aea-ledger-solana/README.md:36]: updated

File: plugins/aea-ledger-solana/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```
+twine upload dist/*
+```
```
[the remark this answers is not in the document]
Comment (solarw) [reply plugins/aea-ledger-solana/README.md:40]: removed

File: plugins/aea-ledger-solana/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Solana crypto plug-in
```
[the remark this answers is not in the document]
Comment (solarw) [reply plugins/aea-ledger-solana/README.md:1]: slightly updated

File: docs/api/configurations/constants.md [hunk 5]
```diff
@@ -4,3 +4,15 @@
 
 Module to declare constants.
 
+<a id="aea.configurations.constants.CONFIG_FILE_TO_PACKAGE_TYPE"></a>
+
+#### CONFIG`_`FILE`_`TO`_`PACKAGE`_`TYPE
+
+type: Dict[str, str]
```
[the remark this answers is not in the document]
Comment (solarw) [reply docs/api/configurations/constants.md:11]: @angrybayblade probably related to the doc generator. confused as well but was done automatically on `make generators`

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 6]
```diff
@@ -0,0 +1,1121 @@
...
+
+def _pako_inflate(data):
+    # https://stackoverflow.com/questions/46351275/using-pako-deflate-with-python
+    decompress = zlib.decompressobj(15)
+    decompressed_data = decompress.decompress(data)
+    decompressed_data += decompress.flush()
+    return decompressed_data
+
+
+class SolanaCrypto(Crypto[Keypair]):
```
[the remark this answers is not in the document]
Comment (solarw) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:82]: @angrybayblade , no, it's solana.KeyPair. it's generic class definition for account/key type.
it's ok

File: docs/api/configurations/constants.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (angrybayblade) [reply docs/api/configurations/constants.md:11]: This was caused by dependency mismatch on the doc check and doc generator, see this commit here [c410473](https://github.com/valory-xyz/open-aea/pull/590/commits/c410473daa3481d3ad3b6aadf96ca1b9042d46c2) and [3ad6af7](https://github.com/valory-xyz/open-aea/pull/590/commits/3ad6af7304a53d843afb578a2c7449c0fc8efd4e)

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 7]
```diff
@@ -0,0 +1,1138 @@
...
+        result = subprocess.run(
+            [cmd], capture_output=True, text=True, shell=True)
+
+        if result.stderr != "":
+            raise ValueError(result.stderr)
+
+        return result.stdout
+        """
+        raise NotImplementedError
+
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/solana.py:959]: Please explain

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 8]
```diff
@@ -0,0 +1,1138 @@
...
+        :param contract_instance: the contract to use
+        :param method_name: the contract method to call
+        :param method_args: the contract call parameters
+
+        # noqa: DAR202
+
+        :return: the call result
+        """
+
+        raise NotImplementedError
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/solana.py:986]: Won't we require this method implementation?

File: .github/workflows/workflow.yml [hunk 9]
```diff
@@ -453,6 +458,8 @@ jobs:
           cd ..
           rm -rf go-ipfs
           ipfs init
+          mkdir ~/.config/solana
+          echo "[6,206,143,203,220,22,227,247,237,23,114,181,56,70,53,98,224,64,192,221,134,115,153,146,252,81,8,41,237,224,225,176,195,11,129,62,207,146,39,47,172,98,101,87,102,166,105,202,33,169,134,76,27,149,148,68,4,15,215,47,196,227,216,48]" > ~/.config/solana/id.json
```
[the remark this answers is not in the document]
Comment (solarw) [reply .github/workflows/workflow.yml:462]: @angrybayblade 
environment setup to run tests

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py (hunk 7 above)
Comment (solarw) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:959]: @angrybayblade this is commented out code from the external developer. 
at the moment I can not find the way to deploy solana contract without app being used.

**Review by angrybayblade — APPROVED** [review]

**Review by angrybayblade — APPROVED** [review]

**Pull request merged**